ganttproject/src/net/sourceforge/ganttproject/task/algorithm/RecalculateTaskCompletionPercentageAlgorithm.java

```diff
@@ -43,29 +43,52 @@ public void run(Task task) {
     recalculateSupertaskCompletionPercentageBottomUp(task, facade);
   }
 
+  private class PropagatedDays {
+    public int myCompletedDays;
+    public long myPlannedDays;
+
+    PropagatedDays() {
+      myCompletedDays = 0;
+      myPlannedDays = 0;
+    };
+
+    PropagatedDays(int completedDays, long plannedDays) {
+      myCompletedDays = completedDays;
+      myPlannedDays = plannedDays;
+    };
+  }
+
   private void recalculateSupertaskCompletionPercentageBottomUp(Task task, TaskContainmentHierarchyFacade facade) {
```

Comment:
If you take the root task at lines 31-35, you can remove this method completely.

Reply:
Ok, but then the implementation of `void run(Task task)` has to be changed too or its invocations need to be replaced by` run()`.